### ISSUE TITLE
Netlify Dev keeps throwing errors when chanigng functions

### ISSUE THREAD

**Issue by timroes** [netlify/cli#2244]
**Describe the bug**

When using `netlify dev` and functions (in TypeScript) editing those functions and saving those files will keep randomly causing issues with netlify dev. Editing a functions file and saving it will always show the following output:

```
◈ Function builder zip-it-and-ship-it building functions from directory /home/netlify-issues/src/server
◈ Function builder zip-it-and-ship-it finished building functions from directory /home/netlify-issues/src/server
```

But requesting data from the API, very often will now result in the following error:

```
Request from ::1: GET /.netlify/functions/test-api
{"level":"error","message":"End - Error:"}
{"errorMessage":"Cannot find module '/home/netlify-issues/.netlify/functions-serve/test-api.js'
Require stack:
- /home/netlify-issues/node_modules/lambda-local/build/lambdalocal.js
- /home/netlify-issues/node_modules/netlify-cli/src/utils/serve-functions.js
- /home/netlify-issues/node_modules/netlify-cli/src/commands/dev/index.js
- /home/netlify-issues/node_modules/@oclif/config/lib/plugin.js
- /home/netlify-issues/node_modules/@oclif/config/lib/config.js
- /home/netlify-issues/node_modules/@oclif/config/lib/index.js
- /home/netlify-issues/node_modules/@oclif/command/lib/command.js
- /home/netlify-issues/node_modules/@oclif/command/lib/index.js
- /home/netlify-issues/node_modules/netlify-cli/src/index.js
- /home/netlify-issues/node_modules/netlify-cli/bin/run","errorType":"Error","stackTrace":["uire stack:","home/netlify-issues/node_modules/lambda-local/build/lambdalocal.js","home/netlify-issues/node_modules/netlify-cli/src/utils/serve-functions.js","home/netlify-issues/node_modules/netlify-cli/src/commands/dev/index.js","home/netlify-issues/node_modules/@oclif/config/lib/plugin.js","home/netlify-issues/node_modules/@oclif/config/lib/config.js","home/netlify-issues/node_modules/@oclif/config/lib/index.js","home/netlify-issues/node_modules/@oclif/command/lib/command.js","home/netlify-issues/node_modules/@oclif/command/lib/index.js","home/netlify-issues/node_modules/netlify-cli/src/index.js","home/netlify-issues/node_modules/netlify-cli/bin/run","Function.Module._resolveFilename (internal/modules/cjs/loader.js:880:15)","Function.Module._load (internal/modules/cjs/loader.js:725:27)","Module.require (internal/modules/cjs/loader.js:952:19)","require (internal/modules/cjs/helpers.js:88:18)","Object._executeSync (/home/netlify-issues/node_modules/lambda-local/build/lambdalocal.js:277:26)","Object.execute (/home/netlify-issues/node_modules/lambda-local/build/lambdalocal.js:83:22)","executeSynchronousFunction (/home/netlify-issues/node_modules/netlify-cli/src/utils/serve-functions.js:102:15)","handler (/home/netlify-issues/node_modules/netlify-cli/src/utils/serve-functions.js:287:12)","Layer.handle [as handle_request] (/home/netlify-issues/node_modules/express/lib/router/layer.js:95:5)","next (/home/netlify-issues/node_modules/express/lib/router/route.js:137:13)"],"level":"error"}
Response with status 500 in 7 ms.
```

Saving the file again (even without changes) often tends to resolve that error, but for me it's around a 50:50 chance if after saving the file this error will appear or not. The error indicates that it looked for the `/.netlify/functions-serve/test-api.js` file. This file does not exist, since my source was under `src/server/test-api/index.js`, which seem to compile into the following folder structure:

```
/.netlify/functions-serve/test-api
- test-api.js
- src/
  - index.js
```

I have not found any triggering indicator for this. It seems to happen randomly, no matter how long I wait after changing the file before requesting the API again, and no matter what the actual change is.

**To Reproduce**

1. Checkout https://github.com/timroes/netlify-issues/tree/issue/cli-rebuilding (simple create react app with one netlify function)
2. Run `yarn install && yarn start` in that folder
3. Load `localhost:8888` which show a simple message loaded from a netlify function.
4. Edit the src/server/test-api/index.ts file in the repository (just remove or add console.logs in there) and save it
5. Refresh the `localhost:8888` page to trigger a new API load.
6. Either see the above failure or if it succeeds, just keep repeating 4 and 5 (happens for me really reliable in the given repository).

**Configuration**

Configuration can be seen in the above mentioned repository.

```bash
> npx envinfo --system --binaries --npmPackages netlify-cli --npmGlobalPackages netlify-cli

  System:
    OS: Linux 5.10 Arch Linux
    CPU: (12) x64 Intel(R) Core(TM) i7-9750H CPU @ 2.60GHz
    Memory: 13.64 GB / 31.01 GB
    Container: Yes
    Shell: 5.8 - /bin/zsh
  Binaries:
    Node: 14.15.5 - ~/.nvm/versions/node/v14.15.5/bin/node
    Yarn: 1.22.10 - ~/netlify-issues/node_modules/.bin/yarn
    npm: 6.14.11 - ~/.nvm/versions/node/v14.15.5/bin/npm
  npmPackages:
    netlify-cli: ^3.21.12 => 3.21.12 
```

**Expected behavior**

Saving a function should "just work" and not cause netlify dev to throw errors and require the function file to repeatatly need to be saved again untill fortune favoured you. :-)

**Comment by eduardoboucas** [comment]
This should also be fixed by version `3.22.0` should fix this. Could you give it a try, please? Thanks!

**Comment by timroes** [comment]
I can confirm this is fixed with 3.22.0.
